app/models/scheme.rb

```diff
@@ -230,4 +210,36 @@ def validate_confirmed
       end
     end
   end
+
+  def available_from
+    created_at
+  end
+
+  def status
+    return :active if deactivation_date.blank?
+    return :deactivating_soon if Time.zone.now < deactivation_date
+
+    :deactivated
+  end
+
+  def active?
+    status == :active
+  end
+
+  def implicit_run_deactivation_validations
+    deactivation_date.present? || @run_deactivation_validations
```

Comment:
Would this work instead? Wouldn't need to pass run_deactivation_validations parameter if so

`deactivation_date.present? || :deactivation_date_type_changed?`

Reply:
I don't think so - deactivation date type wouldn't change if a user resubmits "other" with an empty deactivation_date, but we want the validation to still trigger in that case.

Reply:
I'm not sure what that _changed? method is - do you have a link to some docs?